File: code/generate_strategies/IR(summary)/java/java/function_Schematics_model_to_native.java
package com.example;

import com.google.common.base.Preconditions;

public class Person {
    public String get_native_representation(String data) {
        
        if (data == null || data.isEmpty() || data.split(",").length != 2) {
            return "error: Invalid input format";
        }

        String[] parts = data.split(",", -1);
        String name = parts[0];
        int age;
        try {
            age = Integer.parseInt(parts[1]);
        } catch (NumberFormatException e) {
            return "error: Invalid input format";
        }

        
        try {
            Preconditions.checkNotNull(name, "name is required");
            Preconditions.checkArgument(age >= 0, "age must be greater than or equal to 0");
        } catch (IllegalArgumentException e) {
            return "error: " + e.getMessage();
        }

        
        String city = "Unknown";

        return name + " " + age + " " + city;
    }
}

File: code/generate_strategies/RA(method)/cpp/function_Schematics_model_to_native.cpp
#include <string>
#include <vector>
#include <sstream>
#include <stdexcept>
#include <nlohmann/json.hpp>

using json = nlohmann::json;

class Person {
public:
    std::string get_native_representation(const std::string& data) {
       
        std::vector<std::string> parts;
        std::stringstream ss(data);
        std::string item;
        
        while (std::getline(ss, item, ',')) {
            parts.push_back(item);
        }
        
  
        if (parts.size() != 2) {
            return "error: Invalid input format";
        }
        
    
        std::string name = parts[0];
        int age;
        
     
        try {
            age = std::stoi(parts[1]);
        }
        catch (const std::exception&) {
            return "error: Invalid age format";
        }
        
      
        std::string city = "Unknown";
        
  
        if (name.empty()) {
            return "error: {\"name\": [\"This field is required.\"]}";
        }
        
        if (age < 0) {
            return "error: {\"age\": [\"Value must be greater than or equal to 0.\"]}";
        }
        
      
        return name + " " + std::to_string(age) + " " + city;
    }
};